Makefile

```diff
@@ -0,0 +1,25 @@
+.DEFAULT_GOAL:help
+
+DC=docker-compose -f docker/docker-compose.yml
+
+help:
+	@grep -E '^[a-zA-Z_-]+:.*?## .*$$' $(MAKEFILE_LIST) | sort | awk 'BEGIN {FS = ":.*?## "}; {printf "\033[36m * make %s\033[0m ## %s\n", $$1, $$2}'
+
+init: ## Init config
+	cp config/burrow-cli-config.php.dev config/burrow-cli-config.php
+
+build: ## Build image used to install dependencies and run tests
+	$(DC) build
+
+install: ## Install dependencies with docker
+	$(DC) run -T php composer install
+
+test-integration: ## run integration tests (tests/integration)
+	$(DC) run -T php-amqp-php-lib sh -c "dockerize -wait tcp://rabbitmq:5672 && vendor/bin/phpunit --config=phpunit-integration.xml"
+	$(DC) run -T php-pecl-ext sh -c "dockerize -wait tcp://rabbitmq:5672 && vendor/bin/phpunit --config=phpunit-integration.xml"
+
+test-unit: ## run unit tests (tests/unit)
+	$(DC) run -T php vendor/bin/phpunit --config=phpunit.xml
```

Comment:
Si je clone le projet et que je fais `make test-unit` ça ne marche pas

Il faudrait mettre `vendor` comme dépendance de `test-unit` et définir que vendor se crée en faisant install

Reply:
> Also declare the vendor target to do a composer install

Does it mean that I should rename `install` target to `vendor`. We usually use `install` target to install our dependencies, so I'm not sure it's really compliant with other projects

> vendor should be added as a dependency to test-unit (and other targets that requires vendors to be installed)

I don't think its a good idea to run a `composer install` every time we want to run tests, it'll slow down development. I prefere that you start to run `init` and ` install` first, and then you can run `test-unit`

Reply:
> Does it mean that I should rename install target to vendor

No you shouldn't, you should make `vendor` a dependency to `install` and let `install` empty so it's compliant to other projects.

> I don't think its a good idea to run a composer install every time we want to run tests

It won't be the case. If you define `vendor` as a dependency and a target. Make (the program) will check if the `vendor` folder exists, if not then it will run what has been define as the `vendor` target (so `composer install`).

It is also possible to make `composer.json` a dependency of `vendor` to automatically rerun composer when we pull an update to `composer.json` and local vendors are outdated.

Reply:
I just updated it, is it okay for you ?

Reply:
I just tested I works perfectly :+1: